### PR TITLE
wireless/bcm43xxx: remove global variable for bcmf_sdio_thread

### PR DESCRIPTION
**Pull request by anchao**
## Summary

wireless/bcm43xxx: remove global variable for bcmf_sdio_thread

## Impact

N/A

## Testing

bcm43013 iperf test

### CODE REVIEW

**Review by xiaoxiang781216 — APPROVED** [review]

File: drivers/wireless/ieee80211/bcm43xxx/bcmf_sdio.c [hunk 1]
```diff
@@ -868,7 +865,8 @@ void bcmf_sdio_waitdog_timeout(wdparm_t arg)
 
 int bcmf_sdio_thread(int argc, char **argv)
 {
-  FAR struct bcmf_dev_s *priv = g_sdio_priv;
+  FAR struct bcmf_dev_s *priv = (FAR struct bcmf_dev_s *)
+                                ((uintptr_t)strtoul(argv[1], NULL, 0));
```
Comment (pkarashchenko) [inline drivers/wireless/ieee80211/bcm43xxx/bcmf_sdio.c:869]:
```suggestion
                                ((uintptr_t)strtoul(argv[1], NULL, 16));
```

File: drivers/wireless/ieee80211/bcm43xxx/bcmf_sdio.c [hunk 2]
```diff
@@ -774,11 +769,13 @@ int bcmf_bus_sdio_initialize(FAR struct bcmf_dev_s *priv,
 
   /* Spawn bcmf daemon thread */
 
+  snprintf(arg1, 16, "0x%" PRIxPTR, (uintptr_t)priv);
```
Comment (pkarashchenko) [inline drivers/wireless/ieee80211/bcm43xxx/bcmf_sdio.c:772]: I think `%p` should also give the same result

File: drivers/wireless/ieee80211/bcm43xxx/bcmf_sdio.c (hunk 1 above)
Comment (anchao) [reply drivers/wireless/ieee80211/bcm43xxx/bcmf_sdio.c]: Done

File: drivers/wireless/ieee80211/bcm43xxx/bcmf_sdio.c (hunk 2 above)
Comment (anchao) [reply drivers/wireless/ieee80211/bcm43xxx/bcmf_sdio.c]: Done

**Review by pkarashchenko — APPROVED** [review]
LGTM!

File: drivers/wireless/ieee80211/bcm43xxx/bcmf_sdio.c [hunk 3]
```diff
@@ -774,11 +769,13 @@ int bcmf_bus_sdio_initialize(FAR struct bcmf_dev_s *priv,
 
   /* Spawn bcmf daemon thread */
 
+  snprintf(arg1, 16, "%p", priv);
```
Comment (pkarashchenko) [inline drivers/wireless/ieee80211/bcm43xxx/bcmf_sdio.c:772]: Optional
```suggestion
  snprintf(arg1, sizeof(arg1), "%p", priv);
```

Comment (anchao) [reply drivers/wireless/ieee80211/bcm43xxx/bcmf_sdio.c]: Done

**Pull request merged**